vault/identity_store_entities.go

```diff
@@ -769,15 +820,26 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 		}
 	}
 
+	// Check alias clashes after validating every fromEntity, so that we have a full list of errors
+	if len(aliasClashes) != 0 {
+		var conflictingAliasesInformation string
+		for _, clash := range aliasClashes {
+			clashString := fmt.Sprintf("\nmountAccessor: %s, toEntity ID: %s, fromEntity ID: %s, conflicting toEntity alias ID: %s, conflicting fromEntity alias ID: %s",
+				clash.mountAccessor, clash.toEntityId, clash.fromEntityId, clash.toAliasId, clash.fromAliasId)
+			conflictingAliasesInformation = conflictingAliasesInformation + clashString
+		}
+		return fmt.Errorf("conflicting alias mount accessors between toEntity and fromEntity, clashes:%s", conflictingAliasesInformation), nil
```

Comment:
```suggestion
		return fmt.Errorf("toEntity and at least one fromEntity have aliases with the same mount accessor, repeat the merge request specifying exactly one fromEntity" , clashes:%s", conflictingAliasesInformation), nil
```

Or something like that, I just want the error to be more meaningful to the user.

Reply:
Good suggestion - will update accordingly